tests/Testcontainers.Tests/Unit/Containers/Unix/TestcontainersContainerTest.cs

```diff
@@ -489,6 +490,26 @@ await using (ITestcontainersContainer testcontainer = testcontainersBuilder.Buil
           Assert.EndsWith(name, testcontainer.Name);
         }
       }
+
+      [Fact]
+      public async Task PullPolicyNever()
+      {
+        // Given
+        // An image that actually exists but was not used/pulled previously
+        // If this image is cached/pulled before, this test will fail
+        var previouslyUnusedImage = "alpine:edge";
+
+        // When
+        var testcontainersBuilder = new TestcontainersBuilder<TestcontainersContainer>()
+          .WithImage(previouslyUnusedImage)
+          .WithImagePullPolicy(PullPolicy.Never);
+
+        // Then
+        await using (ITestcontainersContainer testcontainer = testcontainersBuilder.Build())
+        {
+          Assert.ThrowsAsync<DockerImageNotFoundException>(() => testcontainer.StartAsync());
```

Comment:
`await Assert.ThrowsAsync<Exception>(() => testcontainer.StartAsync());`

Reply:
`ThrowsAsync` won't catch `Exception` here since it catches only the **exact** exception type, see: https://stackoverflow.com/a/31686148. So I switched to `ThrowsAnyAsync`.